src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/KeyValuePairModelBinder.cs

```diff
@@ -17,13 +20,29 @@ public class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
     {
         private readonly IModelBinder _keyBinder;
         private readonly IModelBinder _valueBinder;
+        private readonly ILogger _logger;
 
         /// <summary>
-        /// Creates a new <see cref="KeyValuePair{TKey, TValue}"/>.
+        /// <para>This constructor is obsolete and will be removed in a future version. The recommended alternative
+        /// is the overload that also takes an <see cref="ILoggerFactory"/>.</para>
+        /// <para>Creates a new <see cref="KeyValuePair{TKey, TValue}"/>.</para>
         /// </summary>
         /// <param name="keyBinder">The <see cref="IModelBinder"/> for <typeparamref name="TKey"/>.</param>
         /// <param name="valueBinder">The <see cref="IModelBinder"/> for <typeparamref name="TValue"/>.</param>
+        [Obsolete("This constructor is obsolete and will be removed in a future version. The recommended alternative"
+                    + " is the overload that also takes an " + nameof(ILoggerFactory) + ".")]
```

Comment:
Remove extra indentation here, in `SimpleTypeModelBinder` and in `ModelBinderFactory`